### PR TITLE
feat(console): extract total number from headers for useSWR

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Extract "Total-Number" from response header to useSWR's `data`, if `page` and `page_size` is present in url.

Is unnable to get response header in useSWR directly, so we have to "break" data.

## Linear Issue Reference

LOG-1979

## Testing

Local tested.

### CODE REVIEW

File: packages/console/src/hooks/use-swr-fetcher.ts
Comment (demonzoo) [inline packages/console/src/hooks/use-swr-fetcher.ts:16]: I added the dom.d.ts in PR #432 , after that PR is merged, you can use
```
const data = await response.json<ArbitraryObject>();
```

File: packages/console/src/pages/Users/index.tsx
Comment (demonzoo) [inline packages/console/src/pages/Users/index.tsx:113]: What is the last `{' '}`?

File: packages/console/src/pages/Users/index.tsx
Comment (demonzoo) [inline packages/console/src/pages/Users/index.tsx:119]: Should hide this button when it's already the last page. Also, where's the `Previous` button?

File: packages/console/src/hooks/use-swr-fetcher.ts
Comment (demonzoo) [inline packages/console/src/hooks/use-swr-fetcher.ts:26]: if (number === undefined) ?

File: packages/console/src/pages/Users/index.tsx
Comment (demonzoo) [inline packages/console/src/pages/Users/index.tsx:88]: Better use destructuring
```
const [tableRows, totalPageCount] = data;
```

File: packages/console/src/pages/Users/index.tsx
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx:119]: same as above

**Comment by wangsijie** [comment]
Let's hold this PR and wait for Pagination component.

File: packages/console/src/pages/Users/index.tsx
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx]: updated

File: packages/console/src/hooks/use-swr-fetcher.ts
Comment (wangsijie) [reply packages/console/src/hooks/use-swr-fetcher.ts]: updated

File: packages/console/src/pages/Users/index.tsx
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx]: Replaced with "Pagination"

File: packages/console/src/hooks/use-swr-fetcher.ts
Comment (wangsijie) [reply packages/console/src/hooks/use-swr-fetcher.ts]: updated

**Comment by wangsijie** [comment]
Pagination component added, now ready for review.

**Review by demonzoo — APPROVED** [review]

**Pull request merged**
